src/components/dace_interface.ts

```diff
@@ -909,20 +939,36 @@ implements vscode.WebviewViewProvider {
                     return;
                 }
 
-                this.sendPostRequest(
-                    '/get_avg_parallelism',
-                    {
-                        'sdfg': sdfg,
-                    },
-                    (data: any) => {
-                        resolve(data.avgParallelismMap);
-                        DaCeInterface.getInstance()?.hideSpinner();
-                    },
-                    (error: any) => {
-                        this.genericErrorHandler(error.message, error.details);
-                        reject(error.message);
+
+                vscode.window.showInputBox({
```

Comment:
This is quite some code repetition, is it possible to factor this out into a separate small function? (Showing the prompt, checking for the format, showing error message if violated, etc.)